test/test_basic_logic.py

```diff
@@ -1576,3 +1686,34 @@ def test_receiving_goaway_frame_with_unknown_error(self, frame_factory):
         assert c.state_machine.state == h2.connection.ConnectionState.CLOSED
 
         assert not c.data_to_send()
+
+
+def sanity_check_settings_header_data(frames,
+                                      expected_flow_controlled_length,
+                                      expect_padded_flag,
+                                      expected_data_frame_pad_length):
+        """
+        ```frames``` is a list of http2 frames in order of types
+        "settings", "header", and "data", and the ``flags``
+        and ```flow_controlled_length``` of the "data frame" match
+        expectations.
```

Comment:
So, while I was writing a docstring for this function I realised that it is just doing some unnecessary work. I think we should change it so that it doesn't bother to receive the first two frames, as it doesn't do any meaningful validation of them, and it limits the scope of the function to only test straight after connection setup.

We can just ignore the first two frames and assume that they're not relevant in the tests.